src/js/jsx/sections/style/Fill.jsx

```diff
@@ -52,10 +50,31 @@ define(function (require, exports) {
     var Fill = React.createClass({
         mixins: [FluxMixin],
 
-        shouldComponentUpdate: function (nextProps) {
-            return !Immutable.is(this.props.fills, nextProps.fills) ||
-                this.props.index !== nextProps.index ||
-                this.props.readOnly !== nextProps.readOnly;
+        shouldComponentUpdate: function (nextProps, nextState) {
+            return !Immutable.is(this.state.fill, nextState.fill) ||
+                this.props.disabled !== nextProps.disabled;
+        },
+
+        getInitialState: function () {
+            return {
+                layers: Immutable.List(),
+                fill: {}
```

Comment:
That's weird, since even when it's set it's not a bare object. Seems like this should be null?